airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -229,7 +294,26 @@ def get_conn(self) -> paramiko.SSHClient:
             user = self._authorize_os_login(pubkey)
         else:
             user = self.user
-            self._authorize_compute_engine_instance_metadata(pubkey)
+
+            if self._check_running_in_parallel():
+                self.log.info("Several processes are executed in parallel")
+                bucket_name = f"tmp_bucket_{self.project_id}"
+                lock_file_name = "file.lock"
+                lock_path = f"gs://{bucket_name}/{lock_file_name}"
+                self._create_bucket_for_lock(bucket_name=bucket_name)
+
+                max_time_to_wait = 10
+                for time_to_wait in exponential_sleep_generator(initial=1, maximum=max_time_to_wait):
+                    if self._acquire_lock(lock_path=lock_path):
+                        try:
+                            self._authorize_compute_engine_instance_metadata(pubkey)
+                        finally:
+                            self._release_lock(lock_path=lock_path)
+                            break
+                    self.log.info("Another process is running, waiting...")
+                    time.sleep(time_to_wait)
```

Comment:
What happens if this for-loop is over and we never acquired the lock? Should we raise an exception or log warning at least, because in this case the execution continues without calling the `self._authorize_compute_engine_instance_metadata(pubkey)`?

Reply:
Done